
<!-- README.md is generated from README.Rmd. Please edit that file -->

## Omics R Workshop

**Day/Time:** Thursdays 5:30-7:30pm

**Location:** MDCL 3022

**Contact Info:** [gillb3@mcmaster.ca](gillb3@mcmaster.ca)

## Motivation

In the last two decades advancements in high-throughput technologies and
novel bioinformatic tools have enabled the rapid expansion of
omics-based approaches in various fields of research. Such tools can
assist in better characterizing risk factors (e.g genetics),
ascertaining the underlying mechanisms of disease (e.g. transcriptomics,
proteomics, metabolomics) or better defining outcomes of interest. As
such a space to discuss and expand our knowledge of recent developments
in the design and integration of omics studies is quite timely. R is a
programming language for statistical computing, data analysis and
machine learning. With over 12 000 packages available R can support a
large variety of analysis. R is a free and open source programming
language allowing for bugs to be easily detected and fixed, with plenty
of online resources available. R has also become increasingly popular
especially for statistical analysis and data visualization and is a
robust data analysis tool that is the industry-standard for data
science.

## Objectives

The objective of this workshop is to provide weekly hands on tutorials
ranging from beginner to advanced level to help familiarize trainees
with coding languages such as R. Tutorials will focus on how to handle
data in R, as well as how to analyze and visualize various types of data
using examples data sets.

## Workshop Format

We will spend an hour going through the background for the lesson that
week as well as an example data set to apply what you have learned. The
second hour will be used for discussion among peers and troubleshooting
any ongoing projects you may be working on, instructors will also be
available at this time to facilitate.

## Tentative Workshop Breakdown

### Module 1: Introduction to R

- Overview of R Programming
- Downloading and installing R
- The Help function
- Resources and viewing documentation
- Packages in R

  

### Module 2: Learning the basics of programing in R

- R Markdown
- Setup a working directory
- Identify basic data types and assign variables
- Perform basic calculations
- Going over the basic data structures in R and how to work with them
  (i.e. vectors, lists, - matrices, factors, data frames)
- Subsetting
- Writing data
- Reading csv files
- Reviewing excel

  

### Module 3: Basic Stats in R

- Computing basic statistics
- Descriptive Analysis
- Data distribution and transformation
- Comparing means /hypothesis testing
- Covariance and Correlation analysis
- Time series analysis

  

### Module 4: Functions in R

- Loops
- If condition
- The apply family
- Creating and applying functions

  

### Module 5: Handling and managing Data in R

- List management
- Sorting and summarizing variables
- Data transformation
- Aggregating data
- Selecting data
- Merging data frames
- Outlier detection
- Combining vectors

  

### Module 6: Exploratory Data Analysis

- Exploring datasets
- Variation and visualizing distribution
- Missing values
- Data normalization and transformation
- Covariation

  

### Module 7: Visualizing Data in R

- Importing data
- Base R graphics
- Data visualization with ggplot2 package
- Creating different types of plots: bar charts, line graphs,
  histograms, scatter plots, box plots
- Editing/customizing/combining plots
- Using data visualizing for showing statistical models (i.e
  correlation, biplots)
- Other graphical options (i.e. bubble plots, flow diagrams, heatmaps, )

  

### Module 8: Statistical modelling and data mining methods in R

- Regression models
- Unsupervised methods (i.e. types of cluster analysis and PCA)
- Supervised methods (i.e. PLSDA, Random Forest, Neural networks,)

  

### Module 9: Omics workflows

  

## Future Modules:

Future modules will include more advanced workshops looking at using R
as a database, along with working with more specific packages for
transcriptomic, microbiome, metabolomic, and proteomic analysis. In
addition packages such as DIABLO will be explored to demonstrate how
multiple types of omics data can be integrated for more complex
analysis.  
We are happy to take suggestions and invite others to lead a workshop in
more specific research areas.

## Additional Resources

<https://www.youtube.com/marinstatlectures>
<https://r4ds.had.co.nz/workflow-basics.html> <https://r4ds.hadley.nz/>
